---
output:
  rmdformats::html_clean:
    highlight: kate
---


```{r setup, echo=FALSE, cache=FALSE}
library(knitr)
library(rmdformats)

## Global options
options(max.print="75")
opts_chunk$set(cache=TRUE,
               prompt=FALSE,
               tidy=TRUE,
               comment=NA,
               message=FALSE,
               warning=FALSE)
opts_knit$set(width=75)
```

## Microbial Community Analysis Using Chipster (Transcripts for Day 2 Videos)

## Video 1: Tool Overview and Data Importing

### Introduction

Welcome everybody, my name is Jesse Harrison and I work as a data scientist at CSC - IT Center for Science. In this series of four videos I will discuss steps for microbial community analysis using CSC's Chipster platform. The videos will cover tools for data tidying and visualization, as well as statistical tests used for community comparisons. The videos begin from a stage assuming that you have used the Mothur software package to pre-process your data to the point where you have produced a FASTA file with chimeric sequences removed, as well as a Mothur count file and a taxonomy assignment file. 

In this particular video I will go through an overview of different tools and afterwards we will have a closer look at data importing. So let's get started!

### Slide 1

We can begin with a general overview of different steps we can take to process and analyze an amplicon sequencing data set. I will illustrate these using a flow chart providing a high-level summary of different steps that we might typically follow when the goal of the study is to compare the structure and composition of microbial communities.

The first step in our analysis workflow is to create some phyloseq input files using Mothur. Phyloseq is an R package for microbial community analysis that many of the Chipster microbial community analysis tools are based on.

Once we have created a set of suitable input files, we can import those using phyloseq. This results in the creation of a phyloseq object. Any downstream steps as part of the analytical workflow will make use of this object.

After the data have been imported as a phyloseq object, we have various options. There are several tools that provide different summaries and information on the imported data. For example, it is possible to calculate different alpha diversity indices. It is also possible to rarefy the data to even depth. We will discuss rarefying to even depth in the next video, but for now it suffices to know that this is an optional step.

Before further work, such as data transformations and statistics, it is often a good idea to tidy the data, for example by removing low-abundance sequences. Chipster has a number of options for this, including removing singletons and doubletons, and filtering out selected microbial taxa.

Once the data have been tidied, we can move onto data transformation. The data transformation tool is a crucial one to Chipster workflows using microbial community data and we will cover that in detail at a separate stage, in a later video. 

We might, for example, transform our data into a format that is suitable with the DESeq2 R package for differential abundance comparisons. This step also produces a second transformed data set using the variance stabilising transformation available in DESeq2.

It is also possible to perform relative abundance conversions and to visualize the resulting data using bar plots.

Following data transformation, one can compute distance matrices that can be used as the input for multivariate ordinations and significance tests. Multivariate ordinations allow one to visualize variation in microbial community structure using 2D plots. 

Global significance tests can be used to test for overall differences in community structure. If the test results are significant, they can be followed up by further tests. 

Further tests enable pairwise comparisons. In other words, they can be used to determine where exactly the differences revealed by a global test can be found.

Finally, it is possible to perform operational taxonomic unit (or OTU) differential abundance comparisons using DEseq2.

### Slide 2

That was a lot of information! Now let's take a closer look at the steps involved in importing your data using phyloseq. Just to remind you, phyloseq is a broad-use R package that's used in many of the microbial community analysis tools available in Chipster.

So for importing your data using phyloseq, the first step is to generate some phyloseq input files using Mothur. I have an image here in this slide that shows different parameters available for the Generate phyloseq input files tool. When using this tool, there are a number of things we must specify.

### Slide 3

The type of data can correspond to one of two options. The first is 16S, 18S or archaeal data. The other option is internal transcribed spacer (or ITS) data.

The cutoff value corresponds to the percent dissimilarity threshold used for OTU clustering. For example, a cutoff value of 0.03 stands for 3% dissimilarity (or 97% similarity), which is often used as a threshold for OTU clustering.

We also have to specify three input files: a FASTA file with sequences following chimera removal, a count table file and a sequence taxonomy assignment file.

### Slide 4

Running the Generate input files for phyloseq tool produces a number of files:

- A .shared file associated with a phenodata file
- And a consensus taxonomy file

The phenodata file is a table that you can edit using the Chipster interface. It contains unique IDs for each sample and you can also use it to indicate different sample groupings, for example different experimental treatments.

The screenshot I have on this slide shows how a phenodata file is organised. It is arranged as a table with column names on the top and different values entered in each cell. The sample column is on the far left and contains unique IDs for each microbial community. In this example, we also have two columns that have been left empty and two, called chiptype and group, that have been used to indicate group membership. Keep in mind that currently the groups need to be indicated using letters rather than numbers, for example A and B rather than 1 or 2.

The phenodata file is important as that will be used as the basis of steps that come afterwards in the analytical workflow, whether that is for data visualization or statistical analyses.

### Slide 5

Once we have created phyloseq input files and filled in the phenodata file, we can convert the files into a phyloseq object. On this slide I have another screenshot showing different parameters available for the Conver Mothur files into phyloseq object tool.

At the top of the parameter list we have a section saying "Phenodata variable with sequencing sample IDs". This should correspond to that column in the phenodata table that includes unique IDs for each microbial community. 

Underneath that, we have sections for specifying the shared and consensus taxonomy files. When selecting these files, the tool will automatically use the phenodata table associated with the shared file.

### Slide 6

Converting the input files into a phyloseq object will produce two things:

- The phyloseq object, which is stored as a Rda (or a so-called R data) file
- A text summary of the imported object.

The image on the left shows an example text summary of an imported phyloseq object. The top part shows that the phyloseq object is divided into three sections: 

- An OTU table containing information for, in this case, 1114 taxa and a total of 19 different microbial community profiles. This contains OTU abundance information.

- A sample data table with information on sample variables. This includes information on sample groupings and is produced using the phenodata file.

- A taxonomy table with taxonomic affiliations. In this case, the data have been classified using six different taxonomic levels, ranging from Domain to Genus.

The text summary in Chipster also prints out the unique IDs of each community profile and a list of variables used in the phenodata table. These are useful for checking that the data were correctly imported.

The Rda file, then, can be used as the input for the next steps in the analysis workflow, which will be detailed in the next video. Thanks for joining and let's move on to the next part!

## Video 2: Data Inspection and Tidying

### Slide 1

Welcome to Part 2 of this short series on different Chipster tools for microbiome data tidying, visualization and statistical analysis. In this video, we will focus on tools for inspecting and tidying our data, so that we can ensure they are fit for downstream analyses as part of an analytical workflow.

The first tool to consider is one called Sequence numbers, rarefaction curve and alpha diversity estimates. I have two images on this slide, let's start with the one on the right. This image is a screenshot of a text file produced by the tool. 

On the top we have a list of sequence IDs (like F3D0 and F3D1) and associated numbers of sequences. Depending on the community profile, the numbers of sequences range from 2629 and 16662. This type of variation in sequencing depth is typical to microbial community data sets. 

On the bottom of the screenshot on the right, there is a list of alpha diversity estimates. Alpha diversity is a term used to refer to within-community diversity. The first column contains unique IDs (only a smaller section is shown to better display the data on the slide). Moving from left to right, we have several other columns, starting with observed numbers of OTUs. We also have columns for the Chao1 index (which is an estimate of taxon richness based on abundance data), the Shannon diversity index (which takes into account both taxon richness and evenness), and Pielou's evenness. Here, richness pertains to the number of different taxa in a particular community, while evenness pertains to the relative abundance of different taxa within a community. Low evenness means that certain taxa in the community are more dominant than others, while a higher value would correspond to a more even distribution of taxon abundances. Diversity is influenced by both taxon richness and evenness: when richness and evenness increase, generally so does diversity. The tool also provides an option to tabulate some selected phenodata variable together with alpha diversity estimates to make the data easier to read. In this case, the data have been tabulated together with the "group" column in the phenodata table.

Then we have the image on the left, which is a rarefaction curve. This is a line plot with numbers of sequences on the x axis, with increments of 5000 sequences. The values range from zero to over fifteen thousand. The y axis has numbers of observed OTUs, with increments of one hundred. The values range from zero to four hundred. Several lines are shown on the plot, corresponding to different microbial community profiles. These types of plots are used to visualise the relationship between sampling effort (in our case, sequencing depth) and the number of OTUs. If the lines approach the asymptote, this means that there are no more new taxa to discover. In practice, with microbial data sets, that asymptote is very rarely reached. These types of plots are also useful for visualising differences in sequencing depth within a given data set. In our case, certain communities have close to or over 15000 sequences, whereas others have far less.

### Slide 2

Variation in sequencing depth is a source of uncertainty in microbial community analyses and there are many different ways to deal with it. One option that has been traditionally used involves rarefying to even depth, that is, subsampling the data to achieve a common sequencing depth for all community profiles. As you can imagine, this can result in the loss of considerable amounts of data. Rarefying to even depth is possible in Chipster but it should be noted that there is also considerable evidence for drawbacks to this method. If you're interested in finding out more, I would suggest looking up a very often-cited paper published in the journal PLOS Computational Biology in 2014 by Paul McMurdie and Susan Holmes, called Waste Not, Want Not: Why Rarefying Microbiome Data Is Inadmissible. Chipster also offers alternatives for rarefying to even depth. These include particular ways to transform data and internal corrections for unequal library sizes used by the DESeq2 package.   

### Slide 3

The next thing to consider is the available selection of taxon-level clean-up tools. First of all, it is possible to produce an overview of the taxon composition of a particular phyloseq object at some user-specified level (for example, the phylum level). There are also tools for removing undesired sequences from the data set, for example by retaining only those sequences that have been classified as Bacteria or Archaea. It might be surprising but many microbial community data sets also include chloroplast and mitochondrial sequences, and it is possible to remove these if they are not of interest. If, when using the taxon composition overview tool, you discover some particular taxon that you'd like to remove, that is possible by manually entering the name of the taxon you want to exclude. Overall, these functionalities are divided into a set of three different tools.

### Slide 4

Options for tidying microbial community data sets in Chipster further involve tools for visualizing and filtering out low-abundance OTUs. There are two approaches to this: a prevalence-based one and another based on the abundance of singletons and doubletons. Prevalence refers to the number of samples in which a taxon appears at least once. In Chipster it is possible to visualize taxon prevalences and apply a prevalence filter using a user-specified percent threshold. There are also options for tabulating singleton and doubleton abundances as a text summary, and for removing singletons and doubletons from the phyloseq object.

### Slide 5

The figure shown on this slide is a prevalence plot that can be used to visually estimate a suitable prevalence filtering threshold. The x axis shows total read numbers displayed logarithmically, with values of 1, 10, 100 and up to 10000. The y axis corresponds to prevalences shown as fractions in relation to total sample numbers. The y axis values range from 0 to 1 at increments of 0.25, with 1 corresponding to 100%. For example, if a given taxon has a value of 0.05 on the plot, it occurs in 5% of all samples included in the data set. 

Within the plot, there are several panels split based on taxonomic classifications at the phylum level. In this case, proceeding from the top left toward the right and then downward, we have panels for Actinobacteria, OTUs that are unclassified at the phylum level, Bacteroidetes, Deinococcus-Thermus, Firmicutes, Patescibacteria, Proteobacteria, Tenericutes and Verrucomicrobia. Each dot on the plot corresponds to a separate OTU. There is also a dashed line toward the bottom of each panel. This corresponds to a prevalence filtering threshold of 5% that is included in the prevalence plots as a guess. In practice, when performing prevalence filtering, the user may also set other values as the threshold.

## Video 3: Transformations and Ordinations

### Slide 1

Welcome to Part 3 of this video series on different Chipster tools for microbiome data tidying, visualization and statistical analysis. Here we will discuss a topic that is critical to making the most out of microbial community data sets: transformations and ordinations.

While recording this video, Chipster offers four different options for transforming OTU abundance data sets. These are available as part of the Transform OTU counts tool. The screen cap I have on this slide shows different parameters available for the tool. At the top, there is a drop-down menu where we can choose the type of transformation we want to use. That choice is study-specific and the options include:

- Centered log-ratio (CLR) transformation with a pseudocount of one. A pseudocount is used due to microbial community data featuring many zeros.

- Relative abundance conversions, that is, converting OTU counts to proportions.

- Hellinger transformation, which is essentially the square root of relative abundance

- Conversion to the DESeq2 format. This option also produces a separate data set where a variance-stabilising transformation built into the DESeq2 R package has been applied to OTU counts. 

As an input file, the Transform OTU counts tool accepts any phyloseq object saved in the Rda format.

### Slide 2

If you are using the DESeq2 format conversion option, there is an additional parameter that we need to fill in, called Phenodata variable used for DESeq2 conversion. This is because the DESeq2 conversion step specifies a formula upon which the conversion is based. For example, if we were interested in comparing different sample groups as specified under the Phenodata column called "group", we would choose "group" as the variable here.

### Slide 3

Once we have transformed our data, we often want to perform ordinations and statistical tests. While ordinations can be used to visualize differences in microbial community structure between samples, visualizing taxon relative abundances using bar plots can also be helpful as a qualitative method for inspecting potential differences between groups.

Here, as an example, I have an OTU relative abundance bar plot produced by Chipster. The x axis is arranged by unique IDs for each community profile, that is, the sample column in the Phenodata file. The y axis corresponds to relative abundance, with values ranging from zero to one, at increments of 0.25. The value 1 corresponds to 100% relative abundance. The plot has been split into two panels using the "group" column in the phenodata file, resulting in panels called a and b.

This is a stacked bar plot, with each bar showing relative abundances of different phyla. Three different phyla are shown in the plot, corresponding to different colours.

### Slide 4

Ok, next let's move on to distance matrices and ordinations. At the time of recording this video, Chipster offers the possibility to use two different types of distance measures that are used to compute distance matrices: Euclidean distances and Bray-Curtis distances. If a Euclidean distances are determined for data that have been CLR-transformed, there is a separate term for this type of distance measure, and that's called the Aitchison distance. Essentially, a distance matrix in this context is a matrix of similarity comparisons between individual community profiles. This type of matrix is used as the input for many statistical tests that can be used to investigate between-sample differences in community structure.

The Distance matrices and ordinations tool, which is used for distance matrix computation, is also used to produce multivariate ordinations. Two types of ordinations are available: non-metric multidimensional scaling ordinations, or nMDS in short, or distance-based redundancy analyses (db-RDA in short). Just as with other steps in this part of the analytical workflow, the choice of which ordination type to use is study-specific. To find out more about different ordination methods and other topics related to the analysis of microbial community data, I can whole-heartedly recommend a website called "Guide to Statistical Analysis in Microbial Ecology", for which I have provided a link on the slide.

The screenshot I have here shows some of the parameters available for the Distance matrices and ordinations tool. To use the tool, we have to specify the type of distance measure, the type of ordination, and we also need to list that phenodata column which contains unique IDs for each community profile.

### Slide 5

On this slide, I have an example of a non-metric multidimensional scaling ordination. The figure shows individual data points, all of which correspond to different microbial communities, colour-coded using the "group" column in the phenodata file. The individual data points are labelled by their unique IDs.

What ordinations like nMDS do is to attempt condensing multivariate data onto a 2D plot, since 2D images are easier for humans to interpret than 3D plots (not to mention situations involving more than three dimensions). The closer individual data points are to one another, the more similar they are in terms of community structure. In addition to visualizing differences between sample groups, these types of plots can also be useful for identifying outliers in the data. For example, at the top of the plot, there is an individual red point labelled F3D0, which appears somewhat separate from the other data points in group a.

### Slide 6

If we wanted to use a distance-based redundancy analysis instead of nMDS, we also need to specify a particular phenodata variable. The screenshot on this slide shows more of the parameters available in the Distance matrices and ordinations tool. There are sections for specifying whether we want to show unique sample IDs in the resulting plot and which phenodata variable should be used to group the ordination points by either colour or shape. Beneath those, as the last item in the list, there is a section called "Phenodata variable 1 for db-RDA formula specification". If your are planning to use db-RDA in Chipster, this section should be used to indicate a phenodata variable to use for producing the ordination. For example, this could be the variable "group" if we are interested in visualizing potential variation between groups a and b. Those sections are not shown in the screenshot, but it is also possible to specify further phenodata variables when using db-RDA.

### Slide 7

The image here is an example of a distance-based redundancy analysis performed in Chipster. The way to interpret this figure is similar to interpreting an nMDS plot. Different ordination methods will produce different results and, indeed, one of the major decisions one needs to make while analysing microbial community data sets concerns the selection of the most appropriate methods for data visualization and analysis. 

That's it for this video. In the next one, we will have a look at statistical testing.

## Video 4: Statistics

### Slide 1

This video is the fourth and last part of a video series on different Chipster tools for microbiome data tidying, visualization and statistical analysis. Here, we will focus on different statistical tools that can be used for microbial community comparisons.

The first two methods I will discuss are PERMANOVA and PERMDISP. Both are abbreviations, with PERMANOVA standing for "permutational multivariate analysis of variance" and PERMDISP meaning a test for the homogeneity of multivariate dispersions. As you can hear, there is a lot of terminology here and hopefully by the end of this video, concepts such as multivariate dispersion will feel more familiar.

To start with, we can turn our attention toward what these two tests are generally used for and what kinds of questions they can help answer. 

PERMANOVA is available both as a global test and a pairwise test. A global test is used to test for overall differences between groups in a given data set and is our first point of entry. One example of a question this test could help answer is: Does community structure differ between sample groups?

If we find out that such a difference does exist, it is possible to perform pairwise tests that help determine where exactly the difference can be found. A question pairwise PERMANOVA might help answer is: which particular groups differ from one another in terms of microbial community structure? 

It is important to note that PERMANOVA results can be influenced by what we call location effects, as well as dispersion effects. I'll go through the concepts of location and dispersion in more detail in the next slide. For now, though, I can mention that there is a separate test that is used together with PERMANOVA, called PERMDISP. PERMDISP stands for a test for the homogeneity of multivariate dispersions and it is only run if we get a significant PERMANOVA result. Running it can help us find out why exactly the PERMANOVA result is significant.

### Slide 2

In the last slide, I spoke quite a lot about location and dispersion. So the question remains, what do these actually mean in practice? With the help of this slide, I hope to explain these concepts in more detail.

As mentioned, PERMANOVA results can be influenced by both. More specifically, a PERMANOVA result can be significant due to a location effect, a dispersion effect, or a mix of both.

The figure I have here is lifted from a very useful publication by David Warton and colleagues, published in 2011 in the journal Methods in Ecology and Evolution. The slide contains a link to that paper.

The figure illustrates four different location versus dispersion scenarios that we might encounter when running PERMANOVA. There are four plots and each of these corresponds to a multivariate ordination. You might remember that, in Video 3, we discussed how to interpret ordination plots such as the ones shown here.

In each plot, there are two samples indicated by either circle or plus symbol shapes. The x and y axes correspond to arbitrary ordination axes and are shown without units. Each data point corresponds to a given microbial community and the closer the points are to one another, the more similar they are in terms of community structure.

The top left scenario corresponds to a situation where we have no location or dispersion effect. In this case, the data points are evenly distributed in the ordination space and there are no clear differences between the groups, either in terms of where they reside in the space or in terms of how much variation we see within a given group.

On the top right, we have a situation involving a location effect. Data points corresponding to one group are visibly separated from points corresponding to the other group in the ordination space. This is what we refer to as a location effect. However, the two groups are very similar in terms of within-sample variation. In other words, there is no dispersion effect.

On the bottom left, we have no location effect but we have a dispersion effect. Data points from the two groups overlap in the ordination space, but one of the groups, denoted by circle symbols, shows less within-sample variation than the other group, which is more widely distributed. 

Finally, on the bottom right, we have a scenario involving both a location and dispersion effect. The two groups are separated in terms of where they are located in the ordination space, and the group on the left, denoted by circles, shows much less within-sample variation than the other group.

Running PERMANOVA on its own cannot distinguish between location and dispersion effects, which is why it is often accompanied by PERMDISP. Using those two tests together makes it possible to infer whether a significant PERMANOVA result is due to location or dispersion effects, or both.

### Slide 3

On this slide, I have a screenshot of PERMANOVA results generated in Chipster. This is a text summary of statistical test output and there are a few sections that are particularly important for interpreting the outcome.

At the top, under the word "call", there is a row of text corresponding to the formula used to run PERMANOVA. What is probably of most interest, however, is the section under the line saying "Terms added sequentially". Here we have some test results that could, for example, be used as part of a publication or report.

The column Df corresponds to degrees of freedom and provide us with information about the sample size used to perform the analysis. When reporting the results of tests like this, it is customary to list the degrees of freedom for both the variable being tested and the residual degrees of freedom, which here are 1 and 17, respectively. The column called F.model corresponds to the test statistic, which in the case of PERMANOVA is called the pseudo-F value. That value tells us something about the magnitude about the effect we've observed. The column called Pr(>F) corresponds to the statistical significance of the test, or the p value. Conventionally, p values of less than 0.05 are considered statistically significant. In this case, the number reported is 0.001, which in practice means that p is less than 0.001.

### Slide 4

Let's say that we have completed a global test and it turns out statistically significant, as on the previous slide. To find out where exactly the difference or differences lie, it is possible to conduct post-hoc pairwise comparisons.

If we have a significant PERMANOVA result, Chipster offers the option to run pairwise PERMANOVA tests. Conceptually these are similar to global PERMANOVA, but with comparisons performed between individual sample pairs.

If there is a significant PERMDISP result, this can be followed by a Tukey's Honestly Significant Difference (or HSD) test.

Since both approaches involve multiple testing, the test results are corrected for that (which means that sometimes the results can be somewhat conservative).

### Slide 5

The previous methods were all based on distance matrices. Another option for statistical comparisons of microbial communities in Chipster includes using the DESeq2 R package. 

DESeq2 is a method that originates from the RNAseq field, but is increasingly being used my microbiologists. It is rather complex mathematically speaking and I won't cover the details in this video. Instead, we can focus on the types of questions it can help answer and the inferences we can make based on the output. Tests such as PERMANOVA compare microbial communities as a whole, while DESeq2 can help us get down to the level of individual taxa. It can enable us to answer questions like "Which taxa are differentially abundant between samples?", which in turns might allow us to make useful inferences about the data, for example that a particular illness is associated with a decline in the abundance of certain beneficial microbial taxa in the human gut.

While for distance matrix-based tests we'd often use data that have been transformed in some way or another, with DESeq2 the situation is different. Here, we would use untransformed data as the input, as the DESeq2 package features an internal correction for differences in sequencing library size. 

The results after running DESeq2 are given as logfold changes in taxon abundance. I have a useful link up on this slide if any of you would like to find out more detail on DESeq2.

### Slide 6

In the remaining parts of this video, we can focus on how the DEseq2 tool for microbial community comparisons is currently configured in Chipster. Currently, in April 2021, the tool focuses on comparisons of two groups at a time. There may be situations in which the particular phenodata variable we are interested in contains more than two groups and in those situations, it is possible to specify a pair. These are named as Group 1 and Group 2.

Running DESeq2 involves specifying a reference group against which the other group is compared. If comparing two groups, the reference group is selected alphabetically. For example, b would be compared against a, or a group called sick would be compared against a reference group called healthy.

When using a phenodata column containing more than two groups, the reference always corresponds to Group 2. 

### Slide 7

After running the DESeq2 tool, Chipster will produce a large table with logfold changes reported as numbers. However, looking at that alone can be quite challenging if we want to make any quick inferences about the data. Because of that, a summary plot showing logfold changes is also produced, and on this slide I have an example of one.

The x axis shows logfold changes. Logfold changes can be either negative or positive, so in this case the x axis ranges from -8 to 8. The value zero is indicated by a grey line on the plot, to make interpretation of the results easier. On the y axis is a list of different taxa, in this case at the Order level, with the taxonomic classification level being possible to specify by the user. Individual data points in the plot correspond to logfold changes and have been colour-coded according to the class level, with the taxonomic level used here also being possible to specify separately.

This type of plot serves as a useful overview of taxon-level differences between sample groups and it is also possible to download the more detailed results table from Chipster, for further inspection.

That concludes the fourth and final video in this series. Thanks for joining!















